### ISSUE TITLE
Unexpected bad character error in config file

### ISSUE THREAD

**Issue by talal** [twpayne/chezmoi#548]
**Describe the bug**

Hyphen character when used in a key for some data variable in the config file results in error.

I checked and bare-keys with hyphens [are allowed](https://github.com/toml-lang/toml#keys)

**To reproduce**

My `chezmoi.toml`:
```toml
[data]
  personal-address = "me@outlook.com"
  work-address    = "me@company.com"
```

and I have the following in `dot_gitconfig.tmpl`:
```
[user]
	name = My Name
	email = {{ .personal-address }}
```

**Expected behavior**

The template should compile.

**Output of command with `--verbose`**
```
$ chezmoi apply --verbose
chezmoi: template: /Users/talal/.local/share/chezmoi/dot_gitconfig.tmpl:13: unexpected bad character U+002D '-' in command
```

**Output of `chezmoi doctor`**
```
$ chezmoi doctor
     ok: version 1.7.10, commit a402f9eba1245b45de1aa0b05dda4b35c164f345, built at 2019-12-27T22:37:13Z
     ok: runtime.GOOS darwin, runtime.GOARCH amd64
     ok: /Users/talal/.local/share/chezmoi (source directory, perm 700)
     ok: /Users/talal (destination directory, perm 755)
     ok: /Users/talal/.config/chezmoi/chezmoi.toml (configuration file)
     ok: /usr/local/bin/zsh (shell)
     ok: /Users/talal/Sync/Apps/KeePassXC/main.kdbx (KeePassXC database)
     ok: /usr/local/bin/nvim (editor)
     ok: /usr/bin/vimdiff (merge command)
     ok: /usr/local/bin/git (source VCS command, version 2.24.1)
     ok: /usr/local/bin/gpg (GnuPG, version 2.2.19)
warning: op (1Password CLI, not found)
warning: bw (Bitwarden CLI, not found)
warning: gopass (gopass CLI, not found)
     ok: /usr/local/bin/keepassxc-cli (KeePassXC CLI, version 2.5.1)
warning: lpass (LastPass CLI, not found)
warning: pass (pass CLI, not found)
warning: vault (Vault CLI, not found)
```

**Additional context**

I have resolved this by switching to a better structure in my config file:
```toml
[data.email]
  personal = "me@outlook.com"
  work = "me@company.com"
```

and then using `{{ .email.personal }}`. However, I thought it best to bring this to your attention.

**Comment by twpayne** [comment]
Thank you for reporting this and for the very clear description of the problem :)

Variable names in config files must be [Go identifiers](https://golang.org/ref/spec#Identifiers) for compatibility with the [text/template](https://godoc.org/text/template) language that chezmoi uses. This was not specified in chezmoi's documentation. I've updated the documentation and added a check for non-matching names in #549.

**Comment by twpayne** [comment]
Fixed in #549. Thanks again!
